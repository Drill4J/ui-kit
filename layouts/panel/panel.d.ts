/// <reference types="react" />
interface Attrs {
    className?: string;
    align?: 'space-between' | 'end' | 'center';
    verticalAlign?: 'center' | 'stretch' | 'start' | 'end';
    direction?: 'row' | 'column';
    'data-test'?: string;
}
export declare const Panel: import("react").ComponentType<Attrs & DOMAttrs>;
export declare const PanelItem: import("react").ComponentType<{
    className?: string | undefined;
    children?: React.ReactNode;
    onClick: () => void;
} & DOMAttrs>;
export declare const PanelSpread: import("react").ComponentType<{
    className?: string | undefined;
    children?: React.ReactNode;
} & DOMAttrs>;
export {};
