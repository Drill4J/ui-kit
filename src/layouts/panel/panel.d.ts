interface Attrs {
    className?: string;
    align?: 'space-between' | 'end' | 'center';
    verticalAlign?: 'center' | 'stretch' | 'start' | 'end';
    direction?: 'row' | 'column';
    'data-test'?: string;
}
export declare const Panel: DOMComponent<Attrs>;
export declare const PanelItem: DOMComponent<{
    className?: string | undefined;
    children?: React.ReactNode;
    onClick: () => void;
}>;
export declare const PanelSpread: DOMComponent<{
    className?: string | undefined;
    children?: React.ReactNode;
}>;
export {};
