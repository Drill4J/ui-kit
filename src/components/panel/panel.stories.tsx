import React, { FC, useState } from 'react';
import { Args, Meta, Story } from '@storybook/react';
import 'twin.macro';

import { Panel } from './panel';

export default {
  title: 'Panel',
  component: Panel,
} as Meta;

const Template: Story = (args) => (
  <Panel {...args}>
    <Panel.Content>
      <div tw="px-6">
        <h2>Modal content</h2>
        <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Repellendus, iure magni quia excepturi aliquid quas nostrum?</p>
        <p>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quas nisi corrupti dolorem incidunt quisquam.</p>
        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Corrupti ullam praesentium laudantium delectus...</p>
      </div>
    </Panel.Content>
  </Panel>
);

export const Default = Template.bind({});
Default.args = { isDisableFadeClick: false };


export const DisabledFadeClick = Template.bind({});
DisabledFadeClick.args = { isDisableFadeClick: true };
