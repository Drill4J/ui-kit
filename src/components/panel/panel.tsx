/*
 * Copyright 2020 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import 'twin.macro';

interface Props {
  header: React.ReactNode;
  footer?: React.ReactNode;
  children: React.ReactNode;
  isOpen: boolean;
  onClosePanel: () => void;
}

export const Panel = ({
  children, header, isOpen, onClosePanel, footer,
}: Props) => (isOpen
  ? (
    <>
      <div tw="absolute top-0 left-12 z-20 text-monochrome-light-tint text-24 h-full left-12 flex flex-col">
        <div tw="px-6 py-7 leading-32 bg-monochrome-black">{header}</div>
        <div tw="bg-monochrome-black flex-grow" style={{ opacity: '0.97' }}>{children}</div>
        {footer && <div tw="h-18 bg-monochrome-black">{footer}</div>}
      </div>
      <div onClick={onClosePanel} tw="absolute z-10 inset-0 left-12" style={{ background: 'rgba(0, 0, 0, 0.4)' }} />
    </>
  ) : null
);
