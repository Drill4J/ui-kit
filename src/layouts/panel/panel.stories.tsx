import React from 'react';
import { storiesOf } from '@storybook/react';

import { Panel } from './panel';

storiesOf('Panel', module).add('Panel', () => (
  <>
    <h1>Panel Align center</h1>
    <Panel align="center">
      <h1>Vertical align</h1>
      <Panel verticalAlign="center">
        <h2 style={{ backgroundColor: '#007fff' }}>Vertical align</h2>
        <p style={{ backgroundColor: '#00b602' }}>center</p>
      </Panel>
      <h2>Vertical align</h2>
      <Panel verticalAlign="start">
        <h2 style={{ backgroundColor: '#007fff' }}>Vertical align</h2>
        <p style={{ backgroundColor: '#00b602' }}>center</p>
      </Panel>
      <h3>Vertical align</h3>
      <Panel verticalAlign="end">
        <h2 style={{ backgroundColor: '#007fff' }}>Vertical align</h2>
        <p style={{ backgroundColor: '#00b602' }}>center</p>
      </Panel>
    </Panel>
  </>
));
