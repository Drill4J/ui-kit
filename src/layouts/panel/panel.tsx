import { BEM, div } from '@redneckz/react-bem-helper';
import styles from './panel.module.scss';

interface Attrs {
  className?: string;
  align?: 'space-between' | 'end' | 'center';
  verticalAlign?: 'center' | 'stretch' | 'start' | 'end';
  direction?: 'row' | 'column';
  'data-test'?: string;
}

const panel = BEM(styles);

export const Panel = panel(div({ 'data-test': '', style: undefined, onClick: () => {} } as Attrs));
export const PanelItem = panel.item(
  div({ onClick: () => {} } as {
    className?: string;
    children?: React.ReactNode;
    onClick: () => void;
  }),
);
export const PanelSpread = panel.spread(
  div({} as { className?: string; children?: React.ReactNode }),
);
