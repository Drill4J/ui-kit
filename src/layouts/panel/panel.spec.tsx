import React from 'react';
import renderer from 'react-test-renderer';

import { Panel } from './panel';

describe('Panel', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(
      <Panel>
        <span>Panel</span>
        <span>element</span>
      </Panel>
      ,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
