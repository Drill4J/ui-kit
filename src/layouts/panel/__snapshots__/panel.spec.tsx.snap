// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should match snapshot 1`] = `
<div
  className="panel"
  data-test=""
  onClick={[Function]}
>
  <span>
    Panel
  </span>
  <span>
    element
  </span>
</div>
`;
